mlflow/tracking/_model_registry/utils.py

```diff
@@ -25,7 +25,36 @@
 # This means the following combinations are not supported:
 #  - Tracking RestStore & Model Registry RestStore that use different credentials.
 
-# NOTE: SqlAlchemyStore code is commented out here - we can add it back once it's available
+_registry_uri = None
+
+
+def set_registry_uri(uri):
+    """
+    Set the registry server URI.
+
+    :param uri:
+
+                - An empty string, or a local file path, prefixed with ``file:/``. Data is stored
+                  locally at the provided file (or ``./mlruns`` if empty).
+                - An HTTP URI like ``https://my-tracking-server:5000``.
+                - A Databricks workspace, provided as the string "databricks" or, to use a
+                  Databricks CLI
+                  `profile <https://github.com/databricks/databricks-cli#installation>`_,
+                  "databricks://<profileName>".
+    """
+    global _registry_uri
+    _registry_uri = uri
+
+
+def get_registry_uri():
+    """
+    Get the current registry URI.
+
+    :return: The tracking URI.
+    """
+    global _registry_uri
+    return _registry_uri
```

Comment:
One concern I have is that `get_registry_uri` is not semantically equivalent to `get_tracking_uri` which has logic to return default values if there isn't a tracking uri specified through `set_tracking_uri`. For use in `MlflowClient` this is important, because we want to set the registry URI to be:
```
(registry_uri parameter value) or get_registry_uri() or (tracking_uri parameter value) or get_tracking_uri()
``` 
. However, if `get_registry_uri` were to be used in another context as a stand-alone method to get the _final_ registry uri, you'd have to apply the logic^ yourself. If you are familiar with `get_tracking_uri`, you would expect it to do it for you.

One option is to make this function:
```
def get_registry_uri(tracking_uri=None):
    """
    Get the current registry URI.
    :return: The tracking URI.
    """
    global _registry_uri
    return _registry_uri or tracking_uri or get_tracking_uri()
```
though that seems a bit convoluted and replicates the logic for tracking_uri setting which will need to be maintained in sync with other places where we set the tracking_uri (though may be unlikely to change).

Reply:
can we have lower level utility functions that all callers call for tracking and registry URI resolution. something like...

```
def resolve_tracking_uri(tracking_uri=None):
  return tracking_uri or _tracking_uri

def resolve_registry_uri(registry_uri=None, tracking_uri=None):
  return registry_uri or _registry_uri or resolve_tracking_uri(tracking_uri)
```

So for instances, client can call the utility method with provided parameters and utility methods can resolve the ordering.

Reply:
I think there are several uses of `get_tracking_uri` and `get_registry_uri`:
1. For the fluent API user, `get_*_uri` should tell you which tracking or registry server we're currently using.
2. In projects - not relevant for `get_registry_uri`
3. In `MlflowClient`, to resolve the user-provided params with the values set previously.

I think `get_registry_uri` should return considering the fallback options in order to satisfy 1. For 3, it doesn't seem like a common use case, though if someone wants to implement their own version of `MlflowClient` for some reason it'd be useful to have methods to support it. So I propose:

`mlflow/tracking/_model_registry/utils.py`
```
def _get_registry_uri_from_context():
    global _registry_uri
    # in the future, REGISTRY_URI env var support can go here
    return _registry_uri

def get_registry_uri():
    """
    Get the current registry URI specified by `set_registry_uri`.
    :return: The registry URI.
    """
    tracking_uri = get_tracking_uri()
    return _get_registry_uri_from_context() or (tracking_uri if not is_filestore_uri(tracking_uri) else None)    

def _resolve_registry_uri(registry_uri=None, tracking_uri=None):
    return registry_uri or _get_registry_uri_from_context() or _resolve_tracking_uri(tracking_uri)
```

`mlflow/tracking/_tracking_service/utils.py`
```
def _resolve_tracking_uri(tracking_uri=None):
    return tracking_uri or get_tracking_uri()
```

And in `MlflowClient`:
```
def __init__(self, tracking_uri=None, registry_uri=None):
        final_tracking_uri = utils._resolve_tracking_uri(tracking_uri)
        self._registry_uri = registry_utils._resolve_registry_uri(registry_uri, tracking_uri)
```

I kind of want to also refactor the util files so all tracking and registry URI methods are in `tracking/server_uri_utils.py` now that there's a dependency between the two, but it's technically a breaking change so I don't need to do it... or do I :-)

Reply:
cc @aarondav

Reply:
Okkie dokkie! Makes sense to be more careful with the publicly-exposed method. Should still update the docstring :)